include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -181,13 +183,27 @@
 #    define _ONEDPL_PRAGMA_SIMD_ORDERED_MONOTONIC_2ARGS(PRM1, PRM2)
 #endif
 
-#if (__INTEL_COMPILER >= 1900 || !defined(__INTEL_COMPILER) && _ONEDPL_GCC_VERSION >= 40900 || _OPENMP >= 201307)
+#if (__INTEL_LLVM_COMPILER || __INTEL_COMPILER >= 1900 ||                                                              \
+     !defined(__INTEL_LLVM_COMPILER) && !defined(__INTEL_COMPILER) && _ONEDPL_GCC_VERSION >= 40900 ||                  \
+     _OPENMP >= 201307)
 #    define _ONEDPL_UDR_PRESENT 1
 #else
 #    define _ONEDPL_UDR_PRESENT 0
 #endif
 
-#define _ONEDPL_UDS_PRESENT (__INTEL_COMPILER >= 1900 && __INTEL_COMPILER_BUILD_DATE >= 20180626)
+// UDS is broken with Intel(R) oneAPI DPC++ Compiler on Windows
+// It might be fixed in the future releases, which follow 20240000
+#if !defined(_MSC_VER) && __INTEL_LLVM_COMPILER
+#    define _ONEDPL_INTEL_LLVM_UDS_PRESENT 1
+#else
+#    define _ONEDPL_INTEL_LLVM_UDS_PRESENT 0
+#endif
+
+#if _ONEDPL_INTEL_LLVM_UDS_PRESENT || (__INTEL_COMPILER >= 1900 && __INTEL_COMPILER_BUILD_DATE >= 20180626)
+#    define _ONEDPL_UDS_PRESENT 1
+#else
+#    define _ONEDPL_UDS_PRESENT 0
+#endif
```

Comment:
Combine instead as `#if..#elif..#else`?

Also, I prefer to limit the compiler version to the known non-working, as if the next version would work. If it will not, we should see it in testing and increase the limit. It requires more effort, but has less chances to be forgotten once a fix is in place.

Reply:
There are several arguments, why it is better not to limit the version until we know when it is fixed: please, find them in https://github.com/oneapi-src/oneDPL/pull/1182#discussion_r1327230978. Would you still prefer limiting the version?

Reply:
Thanks for pointing out this discussion, it's useful to see the downsides, especially the potential customer impact.

Maybe we can introduce a special function-like macro like this:
```
#define _ONEDPL_BROKEN_AT_VERSION(M, V) (M) 
```
which for testing we can redefine to
```
#define _ONEDPL_BROKEN_AT_VERSION(M, V) (M <= V) 
```

But for now, OK to leave with a TODO.

Reply:
> Combine instead as #if..#elif..#else?

Done

Reply:
> Maybe we can introduce a special function-like macro like this: ...

I will do it in a separate PR and #1191 is a reminder.